core/new-gui/src/app/dashboard/service/user-project/user-project.service.ts

```diff
@@ -0,0 +1,105 @@
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import { Observable, Subject } from "rxjs";
+import { AppSettings } from "../../../common/app-setting";
+import { DashboardWorkflowEntry } from "../../type/dashboard-workflow-entry";
+import { DashboardUserFileEntry } from "../../type/dashboard-user-file-entry";
+import { Project } from "../../type/project";
+import { UserFileService } from "../user-file/user-file.service";
+
+export const PROJECT_BASE_URL = `${AppSettings.getApiEndpoint()}/project`;
+export const USER_PROJECT_LIST_URL = `${PROJECT_BASE_URL}/list`;
+export const WORKFLOW_OF_PROJECT_URL = `${PROJECT_BASE_URL}/workflows`;
+export const FILE_OF_PROJECT_URL = `${PROJECT_BASE_URL}/files`;
+export const RENAME_PROJECT_URL = `${PROJECT_BASE_URL}/update`;
+export const DELETE_PROJECT_URL = `${PROJECT_BASE_URL}/delete`;
+export const CREATE_PROJECT_URL = `${PROJECT_BASE_URL}/create`;
+export const ADD_WORKFLOW_TO_PROJECT_URL = `${PROJECT_BASE_URL}/addWorkflow`;
+export const DELETE_WORKFLOW_FROM_PROJECT_URL = `${PROJECT_BASE_URL}/deleteWorkflow`;
+export const ADD_FILE_TO_PROJECT_URL = `${PROJECT_BASE_URL}/addFile`;
+export const DELETE_FILE_FROM_PROJECT_URL = `${PROJECT_BASE_URL}/deleteFile`;
+
+export const USER_FILE_BASE_URL = `${AppSettings.getApiEndpoint()}/user/file`;
+export const USER_FILE_DELETE_URL = `${USER_FILE_BASE_URL}/delete`;
+
+@Injectable({
+  providedIn: "root"
+})
+export class UserProjectService {
+  private files: ReadonlyArray<DashboardUserFileEntry> = [];
+
+  constructor(private http: HttpClient, private userFileService: UserFileService) { 
+  }
+
+  public retrieveProjectList() : Observable<Project[]>{
+    return this.http.get<Project[]>(`${USER_PROJECT_LIST_URL}`);
+  }
+
+  public retrieveWorkflowsOfProject(pid: number) : Observable<DashboardWorkflowEntry[]>{
+    return this.http.get<DashboardWorkflowEntry[]>(`${WORKFLOW_OF_PROJECT_URL}/` + pid);
+  }
+
+  public retrieveFilesOfProject(pid: number) : Observable<DashboardUserFileEntry[]>{
+    return this.http.get<DashboardUserFileEntry[]>(`${FILE_OF_PROJECT_URL}/` + pid );
+  }
+
+  public getProjectFiles(): ReadonlyArray<DashboardUserFileEntry> {
+    return this.files;
+  }
```

Comment:
I am concerned about this local cache, same reason for the co-editing situation.

Reply:
will resolve usage of this cache in the project-service.ts in future PR to unify duplicate code / logic